pkg/network/setup/configstate_test.go

```diff
@@ -118,36 +125,85 @@ var _ = Describe("config state", func() {
 	})
 
 	It("runs and fails at the config step", func() {
 		injectedErr := fmt.Errorf("fail config")
 		discover, config := &funcStub{}, &funcStub{errRun: injectedErr}
 
-		Expect(configState.Run(testNet, discover.f, config.f)).To(MatchError(injectedErr))
+		Expect(configState.Run(nics, discover.f, config.f)).To(MatchError(injectedErr))
 
-		Expect(discover.executed).To(BeTrue(), "the discover step should execute")
-		Expect(config.executed).To(BeTrue(), "the config step should execute")
+		Expect(discover.executedNICs).To(Equal([]string{testNet0}), "the discover step should execute")
+		Expect(config.executedNICs).To(Equal([]string{testNet0}), "the config step should execute")
 
-		podIFaceCacheDataResult, err := cache.ReadPodInterfaceCache(&baseCacheCreator, uid, testNet)
+		podIFaceCacheDataResult, err := cache.ReadPodInterfaceCache(&baseCacheCreator, uid, testNet0)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(podIFaceCacheDataResult.State).To(Equal(cache.PodIfaceNetworkPreparationStarted))
 	})
+
+	When("with multiple interfaces", func() {
+		BeforeEach(func() {
+			nics = append(nics,
+				podNIC{podInterfaceName: testNet1},
+				podNIC{podInterfaceName: testNet2},
+			)
+		})
+
+		It("runs with no current state (cache is empty)", func() {
+			discover, config := &funcStub{}, &funcStub{}
+
+			Expect(configState.Run(nics, discover.f, config.f)).To(Succeed())
+
+			Expect(discover.executedNICs).To(Equal([]string{testNet0, testNet1, testNet2}))
+			Expect(config.executedNICs).To(Equal([]string{testNet0, testNet1, testNet2}))
+
+			for _, testNet := range []string{testNet0, testNet1, testNet2} {
+				podIFaceCacheDataResult, err := cache.ReadPodInterfaceCache(&baseCacheCreator, uid, testNet)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(podIFaceCacheDataResult.State).To(Equal(cache.PodIfaceNetworkPreparationFinished))
+			}
+		})
+
+		It("runs and fails at the config step, 2nd network", func() {
+			injectedErr := fmt.Errorf("fail config")
+			discover, config := &funcStub{}, &funcStub{errRun: injectedErr, errRunForPodIfaceName: testNet1}
+
+			Expect(configState.Run(nics, discover.f, config.f)).To(MatchError(injectedErr))
+
+			Expect(discover.executedNICs).To(Equal([]string{testNet0, testNet1, testNet2}))
+			Expect(config.executedNICs).To(Equal([]string{testNet0, testNet1}))
+
+			for _, testNet := range []string{testNet0, testNet1, testNet2} {
+				podIFaceCacheDataResult, err := cache.ReadPodInterfaceCache(&baseCacheCreator, uid, testNet)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(podIFaceCacheDataResult.State).To(Equal(cache.PodIfaceNetworkPreparationStarted))
+			}
+		})
+
+	})
 })
 
 type funcStub struct {
-	errRun   error
-	executed bool
+	executedNICs          []string
+	errRun                error
+	errRunForPodIfaceName string
 }
 
-func (f *funcStub) f() error {
-	f.executed = true
-	return f.errRun
+func (f *funcStub) f(nic *podNIC) error {
```

Comment:
nit: `f` is so tiny I didn't noticed it until I saw its definition :see_no_evil: 
How about calling if `Fn` or something similar?

Reply:
This change has not touched the method name.
I also do not see a problem with the minimalist name here, how exactly it hurts reading the code?